# Webscraping 

Webscraping a Wikipedia page and creating a datafarme using only pandas. 

## Code

In [2]:
import pandas as pd

In [9]:
df = pd.read_html('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')[0]

The first table on the Wikipedia page about Toronto Postal Codes was imported.

In [35]:
df_filtered = df[df['Borough'] != 'Not assigned'] 
df_filtered.reset_index(inplace = True)
df_filtered.drop(['index'], axis = 1, inplace = True)
print(df_filtered)

    Postal Code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                         Neighbourhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                            Regent Park, Harbourfront  
3                     Lawrence Manor, Lawrence Heights  
4          Queen's Park, Ontario Provincial Government  
..                                                 ...  
98       The Kingsway, Montgomery Road, Old Mill North  
99                                Church and Wellesley  
100  Business reply mail Processing Centre

C:\Users\maiks\anaconda3\lib\site-packages\pandas\core\frame.py:3997: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  errors=errors,


Not assigned postal codes are dropped from the table. The index of the dataframe has been reset; the old index was dropped.

In [33]:
df_filtered.shape

(103, 3)

The shape of the table is 103 rows and 3 columns.